packages/ui/src/components/LogLine/index.tsx

```diff
@@ -38,15 +39,17 @@ export const LogLine = ({ date, level = 'info', message }: LogLineProps) => {
   return (
     <div className={cx(['group flex w-full justify-between rounded-md px-2 py-1', containerStyle])}>
       <div className="flex items-start gap-4">
-        <p className={cx(['w-36 whitespace-pre text-sm', dateStyle])}>{date.toLocaleString('en-US')}</p>
+        {date && <p className={cx(['w-36 whitespace-pre text-sm', dateStyle])}>{date.toLocaleString('en-US')}</p>}
         <pre className={cx(['whitespace-pre-wrap text-sm', messageStyle])}>{message}</pre>
       </div>
-      <span
-        className="hidden cursor-pointer text-xs text-stone-400 hover:text-stone-800 group-hover:inline dark:text-stone-400 dark:hover:text-stone-200"
-        onClick={() => navigator.clipboard.writeText(message)}
-      >
-        Copy
-      </span>
+      {!hiddenCopy && (
```

Comment:
Let's use a descriptive name instead of double negation:
```suggestion
      {showCopy ? (
```

Reply:
done